tests/common.sh

```diff
@@ -53,7 +53,30 @@ function build_sketches()
         if [ $testcnt -ne $build_rem ]; then
             continue  # Not ours to do
         fi
-        rm -rf $build_dir/*
+
+        if [ -e $cache_dir/core/*.a ]; then
+            # We need to preserve the build.options.json file and replace the last .ino
+            # with this sketch's ino file, or builder will throw everything away.  This
+            # head/tail easier than a complicated SED script to replace the 1 line we need.
```

Comment:
`sed -i "s,^.*sketchLocation.*$,  \"sketchLocation\": \"$sketch\"\,,g" build.options.json`
`touch -r $build_dir build.options.json`

I did not test this `touch` (I expect it will work beyond 2038)

Reply:
Thanks for the SED, I wasn't able to find a good way of escaping the /s in the path OTW!

The touch won't work because the builder regens the gitver.h file after this bit of the script runs.  I've updated it to be 'now + 1 day' which will cover us until the end of time...